### PR TITLE
[NETBEANS-4021] - Support for Jakarta EE 8

### PR DESCRIPTION
**Pull request by juneau001**
This PR contains changes that were made in order to add support for Jakarta EE 8 and Eclipse GlassFish 5.1.

### CODE REVIEW

**Comment by neilcsmith-net** [comment]
@juneau001 what's the status on this?  Looking back through older PRs, and noticed this neither merged or had a milestone attached.  If ready, would be useful if you could squash commits as makes sense, and force push your branch if possible.

**Comment by juneau001** [comment]
@neilcsmith-net I would love to get this PR pushed!  I've had a check failing and I am not sure what to do with it...seems to be Windows related?  I can try to perform the squash commit and force push if you think this CI build check failure is safe to move past for now.

**Comment by neilcsmith-net** [comment]
What check?  Check failing on Travis?  Retriggered the one failure I could see there, although not Windows related?!

Would like to get this in to master in next 24hrs for first beta if we're going to try and include in 12.1.

**Comment by juneau001** [comment]
@neilcsmith-net Yes that failing Travis check seemed to indicate a Windows build failure when I tried to investigate.  I'll look into it again once it completes the process and I will try to move this in 24 hours.  I appreciate you pinging me on this PR.  It would be great to have this in 12.1.

**Comment by juneau001** [comment]
@neilcsmith-net I performed a git rebase -i master on the jakartaee8_support branch.  Seemed to squash all commits in the branch.  Please let me know if this does the trick...thanks!

**Comment by neilcsmith-net** [comment]
@juneau001 thanks.  It's still showing 7 commits here.  I think you need to then rebase on top of HEAD~6 or HEAD~7.  See the page at https://cwiki.apache.org/confluence/display/NETBEANS/git%3A+squash+and+merge (but without the merge to master).  If need be I can do that, but possibly better if you can check everything is correct.

**Comment by juneau001** [comment]
@neilcsmith-net I apologize, but when I try to perform a git log --pretty="%h%x09%ae%x09%s" -n 20, then I only see 3 of my commits at the top, followed by many commits from others.  I tried to perform a rebase on top of HEAD~3 and it opened up the selector for all 176 commits from others and attempts to rebase all.  Again, my apologies but this seems to be more complex possibly because I pushed the PR a few times since March and in the meantime synched my environment to the base to keep current and test.  Anything else I can do to get this PR pushed without possibly hurting things?  Thanks for your time and guidance.

**Comment by neilcsmith-net** [comment]
@juneau001 no worries, I'll see what I can do.

**Comment by neilcsmith-net** [comment]
@juneau001 I've force pushed to your branch so you can double-check before I merge to master.  Did a second checkout and diff and it seems identical.  Think you'd possibly rebased on master without fetching it from upstream?  Seemed slightly out of date - had to resolve a spec version conflict.

**Comment by juneau001** [comment]
@neilcsmith-net Thanks Neil!  I took a look and everything looks correct.  I typically fetch from upstream before I do any work, so I don't believe I rebased on master without fetching.  The issues must have been introduced when I pushed the PR in March, then didn't pick up work again on the PR until summer, where I fetched from master and then made more changes to the branch and re-pushed.

File: nbbuild/antsrc/org/netbeans/nbbuild/extlibs/ignored-overlaps [hunk 1]
```diff
@@ -45,6 +45,7 @@ enterprise/websvc.restlib/external/javax.annotation-api-1.2.jar java/maven.index
 enterprise/javaee.api/external/javax.annotation-api-1.2.jar java/maven.indexer/external/javax.annotation-api-1.2.jar
 enterprise/javaee7.api/external/javax.annotation-api-1.2.jar enterprise/javaee8.api/external/javax.annotation-api-1.2.jar
 enterprise/javaee8.api/external/javax.annotation-api-1.2.jar enterprise/websvc.restlib/external/javax.annotation-api-1.2.jar
+enterprise/jakartaee8.api/external/javax.annotation-api-1.3.5.jar
```
Comment (neilcsmith-net) [inline nbbuild/antsrc/org/netbeans/nbbuild/extlibs/ignored-overlaps:48]: This needs fixing before we can merge. It's causing the verify libs and licenses test to fail. Each line must contain two paths.  Is it meant to be a duplicate of something?  If so, needs to have the other path.  If not, then the whole line needs removing.

Check with `ant verify-libs-and-licenses`

**Comment by neilcsmith-net** [comment]
@juneau001 no problem, but something was awry with history relative to master.  Just need to fix this failing test.  Not sure what the fix is meant to be though.

File: nbbuild/antsrc/org/netbeans/nbbuild/extlibs/ignored-overlaps (hunk 1 above)
Comment (juneau001) [reply nbbuild/antsrc/org/netbeans/nbbuild/extlibs/ignored-overlaps:48]: Thanks I will take a look now and I will be in touch.  I appreciate it.

Comment (juneau001) [reply nbbuild/antsrc/org/netbeans/nbbuild/extlibs/ignored-overlaps:48]: This line needs to be removed and then it repairs the verify-libs-and-licenses.  Do you want me to merge the remote branch into my local and make the change?  I don't want to add more commits if not needed.

Comment (neilcsmith-net) [reply nbbuild/antsrc/org/netbeans/nbbuild/extlibs/ignored-overlaps:48]: I can do it here.  Got it open at the moment.  Just wanted to check why that was added.  Thanks!

**Comment by juneau001** [comment]
@neilcsmith-net Thanks for your assistance with this PR, it is very much appreciated.  I have loaded this PR via git fetch origin pull/2023/head and built it successfully within my environment.  I am unable to find the cause of the Travis build issues, as it is indicating MacOS build failure.  Let me know if you have any thoughts on how to proceed.

**Comment by neilcsmith-net** [comment]
@juneau001 no problem.  The MacOS is GitHub Actions not Travis, and I keep seeing the odd failure with that.  Travis error was something that was green before and again seems to be something that has odd failures.  Going to merge and we can deal with any actual test failure (which I don't think we do have) after beta1. Thanks!

**Pull request merged**